lib/stages/oneDiskVDIFWrite.cpp

```diff
@@ -0,0 +1,338 @@
+		    fprintf(setting_file, "# Warning: The start time is when the program starts it, the time "
+                        "recorded in the packets is more accurate\n");
+		    fclose(setting_file);
+                    INFO("Created setting log file: {:s}\n", setting_log_file);
+		}
+            }    
+        
+	    // input buffer frame size
+	    in_buf_frame_size = in_metadata -> sub_frame_data_size + in_metadata -> sub_frame_metadata_size;
+        }
+        // If not a new file, open a new file.
+	if (!isFileOpen && write_to_disk) {
+	    snprintf(full_path, full_path_len, "%s/%010zu_%02d.%s", data_dir_name,	
+		     file_num, disk_id, file_ext.c_str());
+	    
+	    fd = open(full_path, O_WRONLY | O_CREAT, 0666);
+
+            if (fd == -1) {
+                ERROR("Cannot open file");
+                ERROR("File name was: {:s}", file_name);
+                exit(errno);
+            }
+	    isFileOpen = true;
+	    INFO("Write data to: {:s}", full_path);
+	}
+	out_frame_unix_time = in_metadata -> ctime;
+        add_nsec(out_frame_unix_time, (long)(time_offset * time_res_nsec));	
+	// Record the VDIF file start time
+	if (payload_ctr == 0){
+	    vdif_file_start_time = out_frame_unix_time;
+	}
+	//Start filling the header in the vdif payload
+	assert(out_frame_unix_time.tv_sec > ref_ct.tv_sec && "Reference time is beyond current time.");
+	vdif_header -> seconds = out_frame_unix_time.tv_sec - ref_ct.tv_sec;
+	//INFO("second {:d} {:d}", vdif_header -> seconds, out_frame_unix_time.tv_sec - ref_ct.tv_sec);
+	// Make sure the output frame time is integer of frame resolution + FPGA_start offset.
+	assert((out_frame_unix_time.tv_nsec - intsec_offset) % (vdif_samples_per_frame * time_res_nsec) ==0);
+        vdif_header -> data_frame = out_frame_unix_time.tv_nsec / (vdif_samples_per_frame * time_res_nsec);
+        vdif_header -> thread_id = in_metadata -> freq_start + freq_offset;
+	// copy vdif frequency per frame (4) from input buffer
+	for (uint32_t tt = 0; tt < vdif_samples_per_frame; tt++){
+	    for (uint32_t ii = 0; ii < vdif_freq_per_frame; ii++){
+                // Copy data to payload
+	        // The input buffer is order as freq0(n_sample * n_pol) freq1(n_sample* n_pol) ...
+	        // But the VDIF is order as freq0 (1_sample * n_pol) freq1(1_sample * n_pol) ...
+                src_start = (freq_offset + ii) * in_buf_frame_size + (time_offset + tt) * num_pol  + in_metadata -> sub_frame_metadata_size;
+                target_start = (ii * num_pol) + tt * num_pol * vdif_freq_per_frame;
+	        for (uint32_t jj = 0; jj < num_pol; jj++){
+	            //if (ii >= 0 && ii < 20){		
+	            //    if (tt > 620 && tt < 630 && time_offset > 49000 && time_offset < 51000)
+	   	    //        INFO("target_index {:d} src_index {:d} tt {:d} ii {:d} freq offset {:d} time offset {:d} frameszie {:d}", target_start +jj, src_start + jj, tt, ii, freq_offset, time_offset, in_buf_frame_size);
```

Comment:
You should either make these commented out `INFO` statements into `DEBUG` or `DEBUG2` statements, or remove them entirely.